src/main/scala/co/teapot/tempest/server/TempestDBServer.scala

```diff
@@ -44,19 +43,23 @@ class TempestDBServer(databaseClient: TempestDatabaseClient, config: TempestDBSe
   def nodes(graphName: String, sqlClause: String): util.List[lang.Long] =
     CollectionUtil.toJava(databaseClient.nodeIdsFiltered(graphName, sqlClause))
 
-  /** Returns the type of node reached after k steps along the given edge type. If the edge type
-    * is from nodeType1 to nodeType2, this will return nodeType1 if k is even, or nodeType2 if
-    * k is odd.
-    */
-  def kStepNodeType(edgeType: String, edgeDir: EdgeDir, k: Int): String = {
+
+  def loadEdgeConfig(edgeType: String): EdgeTypeConfig = {
     val edgeConfigFile = new File(config.graphConfigDirectoryFile, s"$edgeType.yaml")
     if (! edgeConfigFile.exists()) {
       throw new UndefinedGraphException(s"Graph config file ${edgeConfigFile.getCanonicalPath} not found")
     }
-    val edgeConfig = ConfigLoader.loadConfig[EdgeTypeConfig](edgeConfigFile)
+    ConfigLoader.loadConfig[EdgeTypeConfig](edgeConfigFile)
+  }
 
-    if ((edgeDir == EdgeDir.Out && k % 2 == 0) ||
-        (edgeDir == EdgeDir.In  && k % 2 == 1)) {
+  /** Returns the type of node reached after k steps along the given edge type. If the edge type
+    * is from nodeType1 to nodeType2, this will return nodeType1 if k is even, or nodeType2 if
+    * k is odd.
```

Comment:
Good call; new comment
```
  /** Returns the type of node reached after k steps along the given edge type starting with the given
    * initial direction. If the edge type
    * is from nodeType1 to nodeType2, and edgeDir is EdgeDirOut, this will return nodeType1 if k is even,
    * or nodeType2 if
    * k is odd.  The parity is swapped if edgeDir is EdgeDirIn.
    */
```